org/mozilla/jss/ssl/javax/JSSEngineReferenceImpl.java

```diff
@@ -1435,21 +1439,45 @@ public void cleanup() {
             session.close();
             session = null;
         }
+
+        cleanupKeys();
+    }
+
+    private void cleanupKeys() {
+        // Clean up the keys used in this transaction.
+        if (cert != null) {
+            try {
+                cert.close();
+            } catch (Exception e) {}
```

Comment:
Could we log a warning if there's an exception so at least we know if there's something wrong?